cloud_storage/cloud_storage/overrides/file.py

```diff
@@ -50,17 +52,74 @@ def on_trash(self) -> None:
 				_("This file is attached to a submitted document and cannot be deleted"),
 				frappe.PermissionError,
 			)
-
+		if len(self.file_association) > 1:
+			frappe.throw(
+				_("This file is attached to multiple documents and cannot be deleted"),
+				frappe.PermissionError,
+			)
 		super().on_trash()
 
+	def associate_files(self) -> File:
+		if not self.content_hash and "/api/method/retrieve" in self.file_url:  # type: ignore
+			associated_doc = frappe.get_value("File", {"file_url": self.file_url}, "name")
```

Comment:
Wouldn't this return the current file?

Reply:
No, the current file doesn't yet have a content hash. This only occurs when associating a second File (doctype) with the file (attachment).